starlette/staticfiles.py

```diff
@@ -154,20 +153,26 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         raise HTTPException(status_code=404)
 
     def lookup_path(
-        self, path: str
-    ) -> typing.Tuple[str, typing.Optional[os.stat_result]]:
+        self, path: Path
+    ) -> typing.Tuple[Path, typing.Optional[os.stat_result]]:
         for directory in self.all_directories:
-            full_path = os.path.realpath(os.path.join(directory, path))
-            directory = os.path.realpath(directory)
-            if os.path.commonprefix([full_path, directory]) != directory:
+            original_path = Path(directory).joinpath(path)
+            full_path = original_path.resolve()
+            directory = Path(directory).resolve()
+            try:
+                full_path.relative_to(directory)
+                is_internal = True
+            except ValueError:
+                is_internal = False
```

Comment:
Path provides `is_relative_to` method but only in 3.9 upwards [here](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to).

Checking the [source](https://github.com/python/cpython/blob/main/Lib/pathlib.py#L732) this is the implementation.

Reply:
The snippet can be simpler  but I think this is more readable. Maybe a personal preference.